March 2024 release notes (#1477)
semgrep/semgrep-docs: docs/release-notes/march-2024.md

  * [<i class="fas fa-external-link fa-xs"></i>1.65.0](https://github.com/semgrep/semgrep/releases/tag/v1.65.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.66.0](https://github.com/semgrep/semgrep/releases/tag/v1.66.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.66.1](https://github.com/semgrep/semgrep/releases/tag/v1.66.1)
  * [<i class="fas fa-external-link fa-xs"></i>1.66.2](https://github.com/semgrep/semgrep/releases/tag/v1.66.2)
  * [<i class="fas fa-external-link fa-xs"></i>1.67.0](https://github.com/semgrep/semgrep/releases/tag/v1.67.0)

## 🌐 Cloud Platform

### Added

### Changed

### Fixed

- Visual Studio Code extension: fixed issue where rules weren't downloaded, leading to no scan results.

## 💻 Code

### Added

- Added support for Python's `yield` keyword, enabling the detection of taint findings from taint sources returned by `yield`.
- Added ability for users to copy file paths displayed in Semgrep Cloud Platform's **Findings** page if they aren't links.
- Added the ability for users to see if there's a version of a rule they're currently using that supports interfile analysis.
- **API**: added ability to get rules metadata from the API.

### Changed

> - Code analysis started by logged-in users running `semgrep ci` now includes cross-file (interfile) analysis.

Reviewer: ```suggestion
- Code analysis started by logged-in users running `semgrep ci` now includes cross-function (intrafile) analysis.
```

- `.phtml` files are now processed as PHP files and analyzed using PHP rules.
- Updated PR comments to include links to specific findings in Semgrep Cloud Platform.

> - Users can see all projects, even if they don't have any identified findings, in Semgrep Cloud Platform.

Reviewer: They should have already been able to see all projects in **Projects**. What's changed here?

> - Semgrep Code now distinguishes between findings resolved by rule changes and findings resolved due to code modifications. This change applies only to new findings.

Reviewer: What's the indication that does this?

Author: I don't think there is one; [here's the conversation for context](https://semgrepinc.slack.com/archives/C04UR6730BX/p1712049874695069)

Reviewer: Ok, i think this is safe to resolve.

>   - Only findings fixed due to code modifications are marked as fixed.

Reviewer: Maybe indent on level inward because it's related to the above?

>     - The fix rate calculated by Semgrep Code now includes only such findings.

Reviewer: Maybe indent on level inward because it's related to the above?

>   - Findings fixed due to rule changes are marked as resolved.

Reviewer: ```suggestion
  - Findings fixed due to rule changes are marked as **resolved**.
```
And also indent one level inward.

### Fixed

- Kotlin: Fixed a parsing error when a newline appears between the class name and the primary constructor.
- Fixed an issue where autofix on variable definitions could not handle semicolons for Java, C++, C#, Rust, Cairo, Solidity, and Dart.
- Fixed an issue with autofix application on lines with multi-byte characters.
- Fixed issue where credentials were inadvertently included in a project URL when publishing a custom rule using `semgrep publish`. Running `semgrep publish` generates a `rule-origin-note`, which includes the project URL in the metadata. When this process occurs in a GitLab CI job, GitLab includes the CI job tokens in the project URL. Semgrep now removes the credential from the metadata.
- Fixed an issue where reachability rules were deleted from Semgrep Registry.
- Fixed an issue where the timestamp on the findings didn't correspond to the timestamp used by the filter; now, both use the `relevant_since` filter, which provides information about when findings were last reopened.

## ⛓️ Supply Chain

### Added

- Supply Chain now offers lockfile-only support for Swift projects.

> - Added a tour of Supply Chain features in Semgrep Cloud Platform for first-time users, as well as a tour for returning users.

Reviewer: I never saw this, even as a returning user :cry: . No action, just an observation.

Reviewer: Actually, this looks like this isn't GA yet, so maybe we want to remove this?

Reviewer: I pulled latest and this was still in, I'll remove it directly on my next commit.

### Changed

- Findings with a critical severity now display in Semgrep Cloud Platform with a darker red color to help distinguish them from high-severity findings.
- Findings are now displayed in Semgrep Cloud Platform with readable names, such as `git-url-parse: Inefficient Regular Expression Complexity` instead of `lodash.defaultsdeep: Improper Input Validation`.

> - Added additional reachability filter values to Semgrep Cloud Platform. Users can now search using the following values: **Reachable**, **Always Reachable**, **Conditionally Reachable**, **Unreachable**, **Unknown**.

Reviewer: ```suggestion
- Added additional reachability filter values to Semgrep Cloud Platform. Users can now search using the following values: **Reachable**, **Always Reachable**, **Conditionally Reachable**, **Unreachable**, **Unknown**.
```

Love that `reachable-if` became conditionally reachable.

Suggest to make these into bullets?

### Fixed

> - Fixed an issue where reachable findings were labeled **Unreachable** in Semgrep Cloud Platform.

Reviewer: Are these the conditionally reachable findings? This bullet needs a little context because it makes it sound like all reachable findings weren't found (because they're unreachable).

Author: I think that might actually have been the case? https://github.com/semgrep/semgrep-app/pull/13351

Reviewer: It looks like this is for the new UI? We can remove if so.

- Fixed an issue where bulk triage didn't work in Semgrep Cloud Platform for Supply Chain findings.
- Fixed an issue where Supply Chain rules and findings erroneously display a confidence label.

## 🤖 Assistant (beta)

### Added

### Changed

### Fixed

## 🔐 Secrets

> Semgrep Secrets is no longer beta. It is now generally available.

Reviewer: ```suggestion
Semgrep Secrets is now generally available.
```
Though - wait a sec - I thought it wasn't yet? [Slack thread](https://semgrepinc.slack.com/archives/C04L9V5NDPG/p1710439729240409)

Reviewer: I see that we removed the beta label, but I don't see a corresponding blog post? We usually have an announcement about it, right?

Author: So we've found that odd in-between, [where secrets is GA but marketing is forthcoming](https://semgrepinc.slack.com/archives/C010CKRBLN9/p1709151677609539)

Reviewer: Let's remove the bullet for now.

### Added

- Historical scanning is now available as a public beta feature. Historical scanning allows users to find valid secrets in their Git commit history. To enable this feature:
    1. Log in to Semgrep Cloud Platform.
    2. Navigate to **Settings** > **Deployments**.
    3. Under **Secrets**, toggle on **Historical scanning**.
  Users can also include the `--historical-secrets` flag when running `semgrep ci` in the CLI.

> - Added the ability to view a Secrets rule if there's one that supersedes a Semgrep Code rule with similar functionality.

Reviewer: ```suggestion
- Added the ability to view a Secrets rule if there's one that supersedes a Semgrep Code rule with similar functionality.
```
Where do users see this?

### Fixed

- Fixed a bug in which users couldn't claim a license if they only had one organization. <!-- 13076 -->
- **Visual Studio Code extension:** fixed an issue where rules weren't downloaded to the user's machine, which resulted in no findings detected.
- Minor UI and in-app copy fixes in the following:
    - Editor
    - Settings page
    - Tutorial page
    - Onboarding process <!-- 13447, 13445, 13364, 13194, 13082 -->
- Fixed a bug in which users were sometimes unable to delete their SSO configuration. <!-- 13171 -->

## 💻 Code

### Added

- Added support for Python's `yield` keyword, enabling the detection of taint findings from taint sources returned by `yield`.
- Added ability for users to copy file paths displayed in Semgrep Cloud Platform's **Findings** page if they aren't links.
- Added the ability for users to see if there's a version of a rule they're currently using that supports interfile analysis.
- Added **Clear filters** button when no findings appear in the **Findings** page after the user has set some filters.
- **API**: added ability to get rules metadata from the API.

### Changed

> - Code analysis started by logged-in users running `semgrep ci` now includes cross-file (intrafile) analysis. This change affects CI jobs and CLI scans.

Reviewer 2: This should say interfile right?

Reviewer: No. It's intr**a**file by default, int**er**file is opt-in. The actual error is "cross-file" should be "cross-function. I'll make the suggestion.

### Fixed

- Fixed a bug in which users couldn't claim a license if they only had one organization. <!-- 13076 -->
- **Visual Studio Code extension:** fixed an issue where rules weren't downloaded to the user's machine, which resulted in no findings detected.
- Minor UI and in-app copy fixes in the following:
    - Editor
    - Settings page
    - Tutorial page
    - Onboarding process <!-- 13447, 13445, 13364, 13194, 13082 -->
- Fixed a bug in which users were sometimes unable to delete their SSO configuration. <!-- 13171 -->

## 💻 Code

### Added

- Added support for Python's `yield` keyword, enabling the detection of taint findings from taint sources returned by `yield`.
- Added ability for users to copy file paths displayed in Semgrep Cloud Platform's **Findings** page if they aren't links.
- Added the ability for users to see if there's a version of a rule they're currently using that supports interfile analysis.
- Added **Clear filters** button when no findings appear in the **Findings** page after the user has set some filters.
- **API**: added ability to get rules metadata from the API.

### Changed

> - Code analysis started by logged-in users running `semgrep ci` now includes cross-file (intrafile) analysis. This change affects CI jobs and CLI scans.

Reviewer: ```suggestion
- Code analysis started by logged-in users running `semgrep ci` now includes cross-function (intrafile) analysis by default. This change affects CI jobs and CLI scans.
```
